### DISCUSSION TITLE
Why is 'aggr=None' not supported for custom MessagePassing modules?

### DISCUSSION THREAD

**Discussion by MasoudMo** [pyg-team/pytorch_geometric#4403] [Ideas]
In the initial argument check for the MessagePassing class, 'None' is accepted as the aggregation scheme:
`assert self.aggr in ['add', 'sum', 'mean', 'min', 'max', 'mul', None]`

However, the scatter function later used to aggregate messages does not allow such an aggregation (aka no aggregation):
```
if reduce == 'sum' or reduce == 'add':
    return scatter_sum(src, index, dim, out, dim_size)
if reduce == 'mul':
    return scatter_mul(src, index, dim, out, dim_size)
elif reduce == 'mean':
    return scatter_mean(src, index, dim, out, dim_size)
elif reduce == 'min':
    return scatter_min(src, index, dim, out, dim_size)[0]
elif reduce == 'max':
    return scatter_max(src, index, dim, out, dim_size)[0]
else:
    raise ValueError
```

Maybe this should be addressed?

**Comment by rusty1s** [comment]
`aggr=None` should only be passed in case you specify your own `aggregate` functionality/override the existing one.

**Reply by MasoudMo** [reply]
Thanks. That makes sense!
